src/components/vm/nics/nicAdd.tsx

```diff
@@ -73,275 +78,224 @@ function getRandomMac(vms: VM[]): string | undefined {
     return undefined;
 }
 
-interface DialogValues extends DialogBodyValues {
-    setNetworkMac: boolean;
-    networkMac: string;
-    permanent: boolean;
+interface NetworkMacValue {
+    set: boolean,
+    val: string,
 }
 
-type OnValueChanged = <K extends keyof DialogValues>(key: K, value: DialogValues[K]) => void;
-
 const NetworkMacRow = ({
-    idPrefix,
-    dialogValues,
-    onValueChanged
+    value,
 } : {
-    idPrefix: string,
-    dialogValues: DialogValues,
-    onValueChanged: OnValueChanged,
+    value: DialogValue<NetworkMacValue>
 }) => {
+    const v_set = value.sub("set");
+    const v_val = value.sub("val");
+
     return (
-        <FormGroup fieldId={`${idPrefix}-generate-mac`} label={_("MAC address")} hasNoPaddingTop isInline>
-            <Radio id={`${idPrefix}-generate-mac`}
-                   name="mac-setting"
-                   isChecked={!dialogValues.setNetworkMac}
-                   label={_("Generate automatically")}
-                   onChange={() => onValueChanged('setNetworkMac', false)} />
-            <Radio id={`${idPrefix}-set-mac`}
-                   name="mac-setting"
-                   isChecked={dialogValues.setNetworkMac}
-                   label={_("Set manually")}
-                   onChange={() => onValueChanged('setNetworkMac', true)} />
-            <TextInput id={`${idPrefix}-mac`}
-                       className="nic-add-mac-setting-manual"
-                       isDisabled={!dialogValues.setNetworkMac}
-                       value={dialogValues.networkMac}
-                       onChange={(_, value) => onValueChanged('networkMac', value)} />
+        <FormGroup
+            label={_("MAC address")}
+            hasNoPaddingTop
+            isInline
+        >
+            <Radio
+                id={v_set.id("off")}
+                name="mac-setting"
+                isChecked={!v_set.get()}
+                label={_("Generate automatically")}
+                onChange={() => v_set.set(false)}
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2435-4e63f3e8-20251209-142520-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)